### PR TITLE
6335 ux feedback

### PR DESCRIPTION
**Pull request by codyseibert**
- changing the mock adminGetUser logic to only throw an error if that user exists in dynamodb.  the mock will check the local dynamodb server for all users and return the user if they exist
- refactoring the createPractitionerUser.js file into multiple persistence methods
- refactor updatePractitionerUserInteractor to invoke these different persistence methods depending on the state of the user (practitioner without a cognito account, practitioner with a cognito account)

### CODE REVIEW

File: shared/src/persistence/dynamo/users/createNewPractitionerUser.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            Name: 'email_verified',
+            Value: 'True',
+          },
+          {
+            Name: 'email',
+            Value: 'practitioner@example.com',
+          },
+          {
+            Name: 'custom:role',
+            Value: ROLES.admissionsclerk,
```
Comment (rachaelparris) [inline shared/src/persistence/dynamo/users/createNewPractitionerUser.test.js]: This doesn't seem good, for the created practitioner to have a role of admissions clerk.

File: web-api/src/applicationContext.js [hunk 2]
```diff
@@ -1323,22 +1331,31 @@ module.exports = (appContextUser, logger = createLogger()) => {
...
             promise: () => {},
           }),
-          adminGetUser: ({ Username = '' }) => ({
-            promise: () => {
-              if (Username.includes('error')) {
-                throw new Error('User does not exist');
-              }
+          adminGetUser: ({ Username }) => ({
+            promise: async () => {
+              // TODO: this scan might become REALLY slow while doing a full integration
```
Comment (rachaelparris) [inline web-api/src/applicationContext.js:1336]: Your PR description says that this was changed to throw an error if the username contains `available`, but this seems like a different set of changes.

File: shared/src/persistence/dynamo/users/createNewPractitionerUser.test.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/persistence/dynamo/users/createNewPractitionerUser.test.js]: good catch!

File: web-api/src/applicationContext.js (hunk 2 above)
Comment (codyseibert) [reply web-api/src/applicationContext.js:1336]: oppps, yeah I describe the change incorrectly!

**Review by rachaelparris — APPROVED** [review]

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**